src/Esfa.Vacancy.Application/Commands/CreateApprenticeship/CreateApprenticeshipParametersMapper.cs

```diff
@@ -17,20 +19,42 @@ public class CreateApprenticeshipParametersMapper : ICreateApprenticeshipParamet
                 ExpectedStartDate = request.ExpectedStartDate,
                 WorkingWeek = request.WorkingWeek,
                 HoursPerWeek = request.HoursPerWeek,
-                WageType = (Domain.Entities.LegacyWageType)request.WageType,
-                LocationTypeId = StandardLocationType, //This should change when more location type are introduced
-                AddressLine1 = request.AddressLine1,
-                AddressLine2 = request.AddressLine2,
-                AddressLine3 = request.AddressLine3,
-                AddressLine4 = request.AddressLine4,
-                AddressLine5 = request.AddressLine5,
-                Town = request.Town,
-                Postcode = request.Postcode,
+                WageType = (LegacyWageType)request.WageType,
+                LocationTypeId = request.LocationType == LocationType.Nationwide ? NationwideLocationType : StandardLocationType,
                 NumberOfPositions = request.NumberOfPositions,
-                VacancyOwnerRelationshipId = employerInformation.VacancyOwnerRelationshipId.Value,
+                VacancyOwnerRelationshipId = employerInformation.VacancyOwnerRelationshipId.Value, //a value should always exist
                 EmployerDescription = employerInformation.EmployerDescription,
                 EmployersWebsite = employerInformation.EmployerWebsite
             };
+
+            MapLocationFields(request, employerInformation, parameters);
+
+            return parameters;
+        }
+
+        private static void MapLocationFields(CreateApprenticeshipRequest request, EmployerInformation employerInformation,
```

Comment:
I suggest breaking this out to a separate class to ease and isolate testing.

Reply:
I see no benefits, as I already have tests to cover all the scenarios.